lib/elixir/lib/kernel.ex

```diff
@@ -2476,7 +2476,7 @@ defmodule Kernel do
         quote do: Module.put_attribute(__MODULE__, unquote(name), unquote(arg),
                                        unquote(stack), unquote(line))
 
-      :lists.member(name, [:moduledoc, :typedoc, :doc]) ->
+      :lists.member(name, [:moduledoc, :typedoc, :doc, :impl]) ->
```

Comment:
It is better to not rely on the line number. That's not a general feature of Elixir module attributes and as such it should be relied on very rarely.